### PR TITLE
Add unit and integration tests to precheck

### PR DESCRIPTION
**Pull request by taquitos**
### Checklist
- [x] I've run `bundle exec rspec` from the root directory to see all new and existing tests pass
- [x] I've followed the _fastlane_ code style and run `bundle exec rubocop -a` to ensure the code style is valid
- [x] I've read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I've updated the documentation if necessary.

### Motivation and Context
_precheck_ launched without tests, lets solve that.

### Description
Added tests for all the critical parts of the tool

### CODE REVIEW

File: precheck/spec/rules/abstract_text_match_rule_spec.rb [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+      return false
+    end
+  end
+
+  describe Precheck do
+    describe Precheck::AbstractTextMatchRule do
+      let(:fail_on_exclusion_rule) { FailOnExclusionTestingTextMatchRule.new }
+      let(:pass_on_exclusion_or_empty_rule) { PassOnExclusionOrEmptyTestingTextMatchRule.new }
+      let(:pass_on_exclusion_rule) { PassOnExclusionTestingTextMatchRule.new }
+      # let(:inclusion_rule) { InclusionTestingTextMatchRule.new() }
```
Comment (KrauseFx) [inline precheck/spec/rules/abstract_text_match_rule_spec.rb]: whops

File: precheck/lib/precheck/rule.rb [hunk 2]
```diff
@@ -5,8 +5,7 @@
 module Precheck
   VALIDATION_STATES = {
     passed: "passed",
-    failed: "failed",
-    skipped: "skipped"
+    failed: "failed"
```
Comment (KrauseFx) [inline precheck/lib/precheck/rule.rb]: Why was this removed?

File: precheck/lib/precheck/rule_processor.rb [hunk 3]
```diff
@@ -28,10 +28,12 @@ def should_trigger_user_error?
 
     def has_errors_or_warnings?
       return true if error_results.length > 0 || warning_results.length > 0
+      return false
     end
 
     def items_not_checked?
       return true if items_not_checked.length > 0
```
Comment (KrauseFx) [inline precheck/lib/precheck/rule_processor.rb]: We could use `return items_not_checked.length > 0`? :)

File: precheck/lib/precheck/rule_processor.rb [hunk 4]
```diff
@@ -28,10 +28,12 @@ def should_trigger_user_error?
 
     def has_errors_or_warnings?
       return true if error_results.length > 0 || warning_results.length > 0
```
Comment (KrauseFx) [inline precheck/lib/precheck/rule_processor.rb]: Same here, could be one line 👍

**Review by KrauseFx — APPROVED** [review]
Wow, these are some great tests, awesome job! 🎷

**Comment by KrauseFx** [comment]
I also added a label + renamed the title for better release notes 👍

File: precheck/lib/precheck/rule.rb (hunk 2 above)
Comment (taquitos) [inline precheck/lib/precheck/rule.rb]: `skipped` wasn't used anywhere. I initially wanted to use it, but opted instead to return nil.

File: precheck/lib/precheck/rule_processor.rb (hunk 3 above)
Comment (taquitos) [inline precheck/lib/precheck/rule_processor.rb]: I think it would need to be `return error_results.length > 0 || warning_results.length > 0` which didn't seem as readable to me.

File: precheck/lib/precheck/rule_processor.rb (hunk 4 above)
Comment (taquitos) [inline precheck/lib/precheck/rule_processor.rb]: Same here, didn't seem as readable.

**Comment by taquitos** [comment]
@KrauseFx best label and rename. Thanks!

File: precheck/lib/precheck/rule_processor.rb (hunk 3 above)
Comment (KrauseFx) [inline precheck/lib/precheck/rule_processor.rb]: ok
